### PR TITLE
BREAKING: convert to lit-element and TypeScript

### LINKED ISSUE #142: Convert to LitElement & TypeScript

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #142

Notes for reviewer:

1. Changed unit tests to not depend on Lumo anymore

2. Moved the following from unit to visual tests:

https://github.com/vaadin/vaadin-tabs/blob/55680b62a1ffaa3846a3879eed824c4e6cc0d0ad/test/nav.html#L168

https://github.com/vaadin/vaadin-tabs/blob/55680b62a1ffaa3846a3879eed824c4e6cc0d0ad/test/nav.html#L191

3. Changed CSS so that with `orientation` set to `undefined` the component looks exactly the same as with `horizontal`. This is a reason why `:not([orientation="vertical"])` is used.

### CODE REVIEW

File: test/visual/anchor-tabs.html [hunk 1]
```diff
@@ -1,26 +1,27 @@
 <!DOCTYPE html>
-
+<html></html>
```
Comment (lkraav) [inline test/visual/anchor-tabs.html]: Hm, this insta-closed `</html>` tag is intentional here, or accidental auto-complete?

Comment (web-padawan) [reply test/visual/anchor-tabs.html]: Thanks, fixed.

**Comment by web-padawan** [comment]
This PR is quite big and I'm going to split it into several parts now as discussed with @yuriy-fix.

There will be several PRs in the following order:

1. `src` files with `test/unit` (with visual tests disabled)
2. `theme` files with `test/visual`
3. README changes, `demo` pages removal etc

**Comment by web-padawan** [comment]
Closing in favor of #144 and #146, will also add remaining changes to another PR.

**Pull request closed without merging**